File: testdata/bench/javascript/BenchmarkTest00009.js
// Vulnerable: SQL injection via knex.raw
const express = require('express');
const knex = require('./knex');
const router = express.Router();
router.post('/test009', (req, res) => {
    const param = req.body.filter;
    knex.raw("SELECT * FROM orders WHERE status = '" + param + "'")
        .then(results => res.json(results))
        .catch(err => res.status(500).json({ error: 'Query failed' }));
});
module.exports = router;

File: testdata/bench/rust/BenchmarkTest00009.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Vulnerable: SQL injection via push_str taint propagation
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("name").unwrap_or(&String::new()).clone();
    let mut sql = String::from("SELECT * FROM users WHERE name = '");
    sql.push_str(&param);
    sql.push_str("'");
    // sqlx::query(&sql).execute(&pool).await
    HttpResponse::Ok().body(sql)
}
